impl/pkg/storage/db/postgres/postgres.go

```diff
@@ -0,0 +1,120 @@
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	"embed"
+	"fmt"
+
+	"github.com/TBD54566975/did-dht-method/pkg/storage/pkarr"
+	pgx "github.com/jackc/pgx/v5"
+	_ "github.com/jackc/pgx/v5/stdlib"
+	goose "github.com/pressly/goose/v3"
+)
+
+//go:embed migrations
+var migrations embed.FS
+
+type postgres string
+
+func NewPostgres(uri string) (postgres, error) {
```

Comment:
public method returns private type!

Reply:
It implements a public interface (`storage.Storage`) that we can't import into this package without creating circular imports. Would you prefer `postgres` type be made public? I definitely see how this can be confusing, but IMO just documenting that the return type implements `storage.Storage` should be sufficient.

Reply:
yes maybe add a note as such, all good